### ISSUE TITLE
Image deployed  does not match helmrelease

### ISSUE THREAD

**Issue by derrickburns** [fluxcd/helm-operator#152]
I got an alarming message from my boss saying that a version of our web site was deployed that has not completed QA testing.  The service in question is installed via a Helmrelease.

Here is the the relevant portion of the `Helmrelease`:
```
apiVersion: helm.fluxcd.io/v1
kind: HelmRelease
metadata:
  annotations:
    fluxcd.io/automated: "true"
    fluxcd.io/tag.blip: glob:release-1.29*
spec:
  chart:
    git: git@github.com:tidepool-org/development
    path: charts/tidepool/0.1.7
    ref: develop
  releaseName: tidepool-production
  rollback:
    enable: true
    force: true
  values:
    blip:
      deployment:
        image: tidepool/blip:release-1.29.0-cecd2bf2a2a95d1b08c722719681e727ac50dcff
        replicas: 2
```

Notably, when I list the workloads using `fluxctl` I see this two workloads for `blip`, one associated with the helmrelease and one associated with the deployment.   
```
WORKLOAD                               CONTAINER       IMAGE                                                               CREATED
production:deployment/blip             blip            tidepool/blip
                                                   |   data-worker-pdf-8d1b9497097d1a7b63dd6175f6c4cb8dd5c0b49b        11 Dec 19 15:35 UTC
                                                   |   data-worker-pdf-3212a01aaef49a8c1020c7d58e6a8c17ffc1e28a        11 Dec 19 15:30 UTC
                                                   '-> release-1.30.0-9a128e60e6b42ccbbee868921b5a7c939db77eab         02 Dec 19 22:54 UTC
                                                       release-1.30.0-c655b727715486262af9055e4b1b15411debaa07         02 Dec 19 22:54 UTC
                                                       new-data-util-0d05851e521be25aeda68ef289b118cbc5c20e91          28 Nov 19 21:06 UTC
production:helmrelease/tidepool        blip            tidepool/blip
                                                   |   data-worker-pdf-8d1b9497097d1a7b63dd6175f6c4cb8dd5c0b49b        11 Dec 19 15:35 UTC
                                                   |   data-worker-pdf-3212a01aaef49a8c1020c7d58e6a8c17ffc1e28a        11 Dec 19 15:30 UTC
                                                   |   release-1.30.0-9a128e60e6b42ccbbee868921b5a7c939db77eab         02 Dec 19 22:54 UTC
                                                   |   release-1.30.0-c655b727715486262af9055e4b1b15411debaa07         02 Dec 19 22:54 UTC
                                                   |   new-data-util-0d05851e521be25aeda68ef289b118cbc5c20e91          28 Nov 19 21:06 UTC
                                                   : (9 image(s) omitted)
                                                   '-> release-1.29.0-cecd2bf2a2a95d1b08c722719681e727ac50dcff         13 Nov 19 05:15 UTC
```

Here is (part of) the deployment that I retrieved using `kubectl`:
```
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  annotations:
    deployment.kubernetes.io/revision: "2"
    flux.weave.works/antecedent: production:helmrelease/tidepool
  creationTimestamp: "2019-12-05T03:03:50Z"
  generation: 4
  labels:
    app: blip
    app.kubernetes.io/instance: tidepool-production
    app.kubernetes.io/managed-by: Tiller
    app.kubernetes.io/name: blip
    helm.sh/chart: blip-0.1.0
  name: blip
  namespace: production
spec:
  replicas: 2
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      app: blip
      app.kubernetes.io/instance: tidepool-production
      app.kubernetes.io/name: blip
  strategy:
    rollingUpdate:
      maxSurge: 25%
      maxUnavailable: 25%
    type: RollingUpdate
  template:
    metadata:
      labels:
        app: blip
        app.kubernetes.io/instance: tidepool-production
        app.kubernetes.io/name: blip
    spec:
      containers:
      - env:
        - name: SKIP_HAKKEN
          value: "true"
        image: tidepool/blip:release-1.30.0-9a128e60e6b42ccbbee868921b5a7c939db77eab
        imagePullPolicy: IfNotPresent
```

So, it appears that `flux` is automating the deployment  of the `blip` service using the `production:deployment/blip` workload while `flux helm operator` is updating `production:helmrelease/tidepool` and those updates are being ignored.

Prior to this happening, I *did* (weeks ago) manually deploy the `blip` service outside of the helmrelease. The deployment did *not* have an `fluxcd.io/automated: "true"` annotation.

I would have expected either:
a) the workload to be managed by the `helmrelease` OR 
b) the workload to be not be automated at all.  

I would *not* expect the `production:deployment/blip` workload to be automated independently of the helmrelease.

**Comment by hiddeco** [comment]
> Notably, when I list the workloads using fluxctl I see this two workloads for blip, one associated with the helmrelease and one associated with the deployment.

This is expected behaviour as `fluxctl` lists what is running in the cluster contradictory to what is in Git.

> So, it appears that flux is automating the deployment of the `blip` service using the `production:deployment/blip` workload while flux helm operator is updating `production:helmrelease/tidepool` and those updates are being ignored.

This is technically impossible, Flux only automates things it has YAMLs for, and only applies (or rather pulls and applies) them on the cluster after pushing the change to Git. As this deployment is the result of a Helm release, there is no YAML in Git to update, push, pull, or apply.

> ```
> ...
> values:
>  blip:
>    deployment:
>      image: tidepool/blip:release-1.29.0-cecd2bf2a2a95d1b08c722719681e727ac50dcff
>      replicas: 2
> ```

This `spec.values` example is [one level too deep for Flux to understand](https://docs.fluxcd.io/en/stable/references/helm-operator-integration.html#automated-image-detection), and will not work without [mapping the paths to the image values](https://docs.fluxcd.io/en/stable/references/helm-operator-integration.html#annotations).

Concluding all of the above, my suspicion is that some change to the chart happened and Flux nor the Helm operator was the perpetrator.

**Comment by derrickburns** [comment]
As I stated above:

Prior to this happening, I did (weeks ago) manually deploy the blip service outside of the helmrelease. The deployment did not have an fluxcd.io/automated: "true" annotation

Also the helmrelease does have an annotation that identifies the location of the image tag. So your comment about reaching down does not apply.

We are a non-profit health care company with real lives at stake when we make mistakes. 

There was no manual deployment of the image that was running. No one edited the image tag. Our config repo lists the correct images. However fluxctl lists a workload that points to the wrong image. 

If we can’t get to root cause, we will have no choice but to discontinue use of flux and flux helm operator in our production cluster.
